### PR TITLE
Fixed issue with Ammo Counts showing 0 when configuring from Lobby

### LINKED ISSUE #6831: [Issue] [0.50.05-SNAPSHOT] Unit losing all ammo on configuration in lobby
### Brief Description *

Opening the configure unit window on a unit in the lobby is setting all ammo counts to 0. If you do not open the configure window the unit will deploy to the match with the default ammo load out.

### 3. Steps to Reproduce

1. Add a unit in the lobby
2. Open configure unit window
3. Observe that on the Equipment  tab, all ammo is defaulting to 0.

### Attach Files

_No response_

### Severity *

High (Major Disruption): A major feature is broken or incorrect, but a workaround exists.

### MegaMek Suite Version *

0.50.05-SNAPSHOT 

### Operating System *

Windows 11 10.0

### Java Version *

Adoptium 17

### Final Verification

- [x] I confirm this is a single, unique issue that hasn't been reported before
- [x] I have included all necessary information and files to help reproduce this issue
- [x] I have asked on MegaMek Discord about this issue
- [x] I have confirmed this issue is being opened on the correct repository: MegaMek, MegaMekLab, or MekHQ

### PR DESCRIPTION
**Pull request by rjhancock**
Fixes #6831 where Ammo Counts would be reset to 0. Bug was caused during general code clean up by me.

Also fixes several other NPE's that showed up during testing related to CF and Entities being null.

### CODE REVIEW

**Review by Sleet01 — APPROVED** [review]
LGTM

File: megamek/src/megamek/common/Mounted.java [hunk 1]
```diff
@@ -766,28 +766,25 @@
...
      */
-    public static int getNumShots(WeaponType wtype, EquipmentMode mode,
-            boolean ignoreMode) {
+    public static int getNumShots(WeaponType wtype, EquipmentMode mode, boolean ignoreMode) {
         // figure out # of shots for variable-shot weapons
-        if (((wtype.getAmmoType() == AmmoType.T_AC_ULTRA) || (wtype
-                .getAmmoType() == AmmoType.T_AC_ULTRA_THB))
-                && (ignoreMode || mode.equals(Weapon.MODE_UAC_ULTRA))) {
+        if (((wtype.getAmmoType() == AmmoType.T_AC_ULTRA) || (wtype.getAmmoType() == AmmoType.T_AC_ULTRA_THB)) &&
+                  (ignoreMode || mode.equals(Weapon.MODE_UAC_ULTRA))) {
```
[the remark this answers is not in the document]
Comment (HoneySkull) [reply megamek/src/megamek/common/Mounted.java:775]: Does this potential null reference need to be addressed?

File: megamek/src/megamek/common/Mounted.java [hunk 2]
```diff
@@ -766,28 +766,25 @@
...
-                || (wtype.getAmmoType() == AmmoType.T_AC_IMP)
-                || (wtype.getAmmoType() == AmmoType.T_PAC))
-                && wtype.hasModes()
-                && (ignoreMode || mode.equals(Weapon.MODE_AC_RAPID))) {
+        else if (((wtype.getAmmoType() == AmmoType.T_AC) ||
+                        (wtype.getAmmoType() == AmmoType.T_LAC) ||
+                        (wtype.getAmmoType() == AmmoType.T_AC_IMP) ||
+                        (wtype.getAmmoType() == AmmoType.T_PAC)) &&
+                       wtype.hasModes() &&
+                       (ignoreMode || mode.equals(Weapon.MODE_AC_RAPID))) {
```
[the remark this answers is not in the document]
Comment (HoneySkull) [reply megamek/src/megamek/common/Mounted.java:784]: Does this need to be addressed?

[the remark this answers is not in the document]
Comment (rjhancock) [reply megamek/src/megamek/common/Mounted.java:784]: I'm addressing them regardless. They need to be dealt with.

File: megamek/src/megamek/common/Mounted.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (rjhancock) [reply megamek/src/megamek/common/Mounted.java:775]: I'm addressing them regardless. They need to be dealt with.
